### PR TITLE
Fix EmptyLineSeparator for multiple single type variables on line

### LINKED ISSUE #8957: EmptyLineSeparator flags commas for multiple single type variables on line
**Check documentation:** https://checkstyle.org/config_whitespace.html#EmptyLineSeparator

```

/var/tmp $ javac Test.java

/var/tmp $ cat Test.java
public class Test {
    int readOnly;
    int p01;
    int p11;
    int q01, q02, q03, q04, q05, q06, q07, q08;
 }


/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="EmptyLineSeparator">
      <property name="allowNoEmptyLineBetweenFields"
                value="true" />
    </module>
  </module>
</module>

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.36.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/temp/Test.java:5:10: ',' should be separated from previous statement. [EmptyLineSeparator]
[ERROR] /var/temp/Test.java:5:15: ',' should be separated from previous statement. [EmptyLineSeparator]
[ERROR] /var/temp/Test.java:5:20: ',' should be separated from previous statement. [EmptyLineSeparator]
[ERROR] /var/temp/Test.java:5:25: ',' should be separated from previous statement. [EmptyLineSeparator]
[ERROR] /var/temp/Test.java:5:30: ',' should be separated from previous statement. [EmptyLineSeparator]
[ERROR] /var/temp/Test.java:5:35: ',' should be separated from previous statement. [EmptyLineSeparator]
[ERROR] /var/temp/Test.java:5:40: ',' should be separated from previous statement. [EmptyLineSeparator]
Audit done.
```

**Expected:** No violations.

### PR DESCRIPTION
**Pull request by strkkk**
Closes #8957

Diff Regression projects: https://gist.githubusercontent.com/strkkk/5eb3b0069ecd438af73bf045c24a855c/raw/d568ee6c30a91e3f721c37b4883ef3264b6203b7/all-projects
Diff Regression config: https://gist.githubusercontent.com/strkkk/bf10f326cacdf50f8cb67f0023cd972b/raw/0dffdf53db3b7cfd7f57c88618fe871f4f779adf/lambda_body.xml

### CODE REVIEW

**Comment by strkkk** [comment]
Github, generate report

**Comment by strkkk** [comment]
travis failed because of infra issue.

@romani for some reason, I do not have rights to restart travis jobs, can you plase add rights for it?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java [hunk 1]
```diff
@@ -611,8 +612,8 @@ private void processVariableDef(DetailAST ast, DetailAST nextToken) {
      */
     private boolean isViolatingEmptyLineBetweenFieldsPolicy(DetailAST detailAST) {
         return detailAST.getType() != TokenTypes.RCURLY
-                && (!allowNoEmptyLineBetweenFields
-                    || detailAST.getType() != TokenTypes.VARIABLE_DEF);
+                && !(allowNoEmptyLineBetweenFields
+                    && TokenUtil.isOfType(detailAST, TokenTypes.COMMA, TokenTypes.VARIABLE_DEF));
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java:608]: Can we keep negatiion close to each Expression ?
It is much easier to read code when  negation is as close as possible.

I need vote on this from maintainers.

**Review by romani — CHANGES_REQUESTED** [review]
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/8957_emptyLineSeparator_2020120511/reports/diff/guava/index.html#A1

It is not a violation between fields. Can we keep it ?

Ok,I see that is grey area of Check https://checkstyle.org/config_whitespace.html#EmptyLineSeparator , it is not declared as target if Check.
Let's keep no violation on such case, but please extend xdoc for this property to show that it is expected.

Item to vote:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java:608]: @pbludov , @nmancus1 , @esilkensen , @rdiachenko , please share your thoughts on negation placing.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorSingleTypeVariables.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+package com.puppycrawl.tools.checkstyle.checks.whitespace.emptylineseparator;
+
+/*
+ * Config:
+ * allowNoEmptyLineBetweenFields = true
+ */
+
+public class InputEmptyLineSeparatorSingleTypeVariables {
+
+    int a; // ok
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorSingleTypeVariables.java:29]: Please add field on line 28, to show it not a violation for Check.

Please recheck that we have such input for default config and it is not a violation also.

**Review by romani — CHANGES_REQUESTED** [review]
Item:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java (hunk 1 above)
Comment (nmancus1) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java:608]: > please share your thoughts on negation placing.

I think that negation of each expression is more clear. in this case.

> If we can give good name .. we can move expression in parentheses in special method.

`isVariableDeclarationList()` ?

Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java]: done, condition is changed, xdoc extended

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorSingleTypeVariables.java (hunk 2 above)
Comment (strkkk) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/InputEmptyLineSeparatorSingleTypeVariables.java:28]: done, I also added example to default config

**Review by romani — APPROVED** [review]

**Review by nmancus1 — APPROVED** [review]
Looks good, @strkkk; the usage of `TokenUtil.isOfType()` really enhances readability.

**Review by pbludov — APPROVED** [review]

**Pull request merged**